gym/utils/env_checker.py

```diff
@@ -384,11 +387,21 @@ def check_env(env: gym.Env, warn: bool = True, skip_render_check: bool = True) -
     # Define aliases for convenience
     observation_space = env.observation_space
     action_space = env.action_space
-    try:
-        env.step(env.action_space.sample())
 
-    except AssertionError as e:
-        assert str(e) == "Cannot call env.step() before calling reset()"
+    # As environments are wrapped by OrderEnforcing generated by `gym.make`
+    if has_wrapper(env, OrderEnforcing):
+        try:
+            env.step(env.action_space.sample())
+        except ResetNeeded as e:
+            assert str(e) == "Cannot call env.step() before calling env.reset()"
+        try:
+            env.render()
+        except ResetNeeded as e:
+            assert (
+                str(e) == "Cannot call env.render() before calling env.reset(), "
+                "if this is a intended action, set `disable_render_order_enforcing` "
+                "on the OrderEnforcer wrapper = False."
+            )
```

Comment:
Do we assume that the environment instance is completely fresh when it is passed to this method? In that case we would also want to assert that an exception is actually thrown. That is not being done atm.

Reply:
That is a good point, I will change to `pytest.raises`

Reply:
Yeah I agree. That should be done in a unit test of the OrderEnforcing wrapper

Reply:
Cool, I have removed it from the env checking